docs/feature_programmable_button.md

```diff
@@ -0,0 +1,67 @@
+## Programmable Button
+
+Programmable button is a feature that can be used to send keys that have no
+predefined meaning.
+This means they can be processed on the host side by custom software without
+colliding without the operating system trying to interpret these keys.
+
+### Enabling Programmable Button support
+
+To enable Programmable Button, add the following line to your keymap’s `rules.mk`:
+
+```c
+PROGRAMMABLE_BUTTON_ENABLE = yes
+```
+
+### Mapping
+
+In your keymap you can use the following keycodes to map key presses to Programmable Buttons:
+
+|Key                       |Description                                                     |
+|--------------------------|----------------------|
+|`KC_PROGRAMMABLE_BUTTON1` |Programmable button 1 |
```

Comment:
Since these keycodes are not "basic" keycodes, they should not use the `KC_` prefix - I would suggest something like `PROGRAMMABLE_BUTTON_x` with short aliases (<= 7 chars) `PB_x`.

Reply:
Done